lib/util/id3_utils.js

```diff
@@ -0,0 +1,298 @@
+            'encoding');
+        return null;
+      }
+      const descriptionEndIndex = frame.data.subarray(1).indexOf(0);
+
+      if (descriptionEndIndex === -1) {
+        return null;
+      }
+      const description = StringUtils.fromUTF8(
+          BufferUtils.toUint8(frame.data, 1, descriptionEndIndex));
+      const data = StringUtils.fromUTF8(
+          BufferUtils.toUint8(frame.data, 2 + descriptionEndIndex));
+
+      metadataFrame.description = description;
+      metadataFrame.data = data;
+      metadataFrame.value = metadataFrame.data;
+      return metadataFrame;
+    } else if (frame.type === 'WXXX') {
+      /*
+       * Format:
+       * [0]   = {Text Encoding}
+       * [1-?] = {Description}\0{URL}
+       */
+      if (frame.size < 2) {
+        return null;
+      }
+      if (frame.data[0] !== shaka.util.Id3Utils.UTF8_encoding) {
+        shaka.log.warning('Ignore frame with unrecognized character ' +
+            'encoding');
+        return null;
+      }
+      const descriptionEndIndex = frame.data.subarray(1).indexOf(0);
+
+      if (descriptionEndIndex === -1) {
+        return null;
+      }
+      const description = StringUtils.fromUTF8(
+          BufferUtils.toUint8(frame.data, 1, descriptionEndIndex));
+      const data = StringUtils.fromUTF8(
+          BufferUtils.toUint8(frame.data, description.length + 1));
+
+      metadataFrame.description = description;
+      metadataFrame.data = data;
+      metadataFrame.value = metadataFrame.data;
+      return metadataFrame;
+    } else if (frame.type === 'PRIV') {
+      /*
+       * Format: <text string>\0<binary data>
+       */
+      if (frame.size < 2) {
+        return null;
+      }
+      const ownerEndIndex = frame.data.indexOf(0);
+      if (ownerEndIndex === -1) {
+        return null;
+      }
+      const owner = StringUtils.fromUTF8(
+          BufferUtils.toUint8(frame.data, 0, ownerEndIndex));
+      const data = BufferUtils.toArrayBuffer(
+          frame.data.subarray(owner.length + 1));
+      metadataFrame.description = owner;
+      metadataFrame.data = data;
+      metadataFrame.value = metadataFrame.data;
+      return metadataFrame;
+    } else if (frame.type[0] === 'T') {
+      /*
+       * Format:
+       * [0]   = {Text Encoding}
+       * [1-?] = {Value}
+       */
+      if (frame.size < 2) {
+        return null;
+      }
+      if (frame.data[0] !== shaka.util.Id3Utils.UTF8_encoding) {
+        shaka.log.warning('Ignore frame with unrecognized character ' +
+            'encoding');
+        return null;
+      }
+      const text = StringUtils.fromUTF8(frame.data.subarray(1));
+      metadataFrame.data = text;
+      metadataFrame.value = metadataFrame.data;
+      return metadataFrame;
+    } else if (frame.type[0] === 'W') {
+      /*
+       * Format:
+       * [0-?] = {URL}
+       */
+      const url = StringUtils.fromUTF8(frame.data);
+      metadataFrame.data = url;
+      metadataFrame.value = metadataFrame.data;
```

Comment:
Why are data and value always the same?  Why have two fields for the same thing?

Reply:
This is for compatibility with the same thing mux.js uses, but I can do what you want

Reply:
I'm going to try to change it, but first I want to unify it with how Safari works. But yeah, it's one thing we shouldn't do.

Reply:
Changed!